test/OpenTelemetry.Instrumentation.AspNetCore.Tests/InProcServerTests.cs

```diff
@@ -14,7 +14,7 @@
 // limitations under the License.
 // </copyright>
 
-#if NET6_0_OR_GREATER
+#if NET6_0 || NET7_0
```

Comment:
Same question as above. Any idea what happened here? Seems kind of odd something would change for .NET 8.

Reply:
@CodeBlanch: 
I've updated the code to only not run when !OSX || !NET8_0.
https://github.com/open-telemetry/opentelemetry-dotnet/issues/4437#issuecomment-1741563896
I'm still trying to understand why.

Reply:
What error did you see if we don't make the change? @Yun-Ting

Reply:
2023-09-29T22:56:50.2880408Z [xUnit.net 00:00:01.49]         /home/runner/work/opentelemetry-dotnet/opentelemetry-dotnet/test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs(1001,0): at OpenTelemetry.Instrumentation.AspNetCore.Tests.BasicTests.DiagnosticSourceExceptionCallBackIsNotReceivedForExceptionsHandledInMiddleware()
2023-09-29T22:56:50.2888622Z [xUnit.net 00:00:01.49]         --- End of stack trace from previous location ---
2023-09-29T22:56:50.4395968Z warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
2023-09-29T22:56:50.4396807Z       Failed to determine the https port for redirect.
2023-09-29T22:56:51.2553080Z   Failed OpenTelemetry.Instrumentation.AspNetCore.Tests.BasicTests.DiagnosticSourceExceptionCallBackIsNotReceivedForExceptionsHandledInMiddleware [204 ms]
2023-09-29T22:56:51.2554395Z   Error Message:
2023-09-29T22:56:51.2555049Z    Assert.Equal() Failure: Values differ
2023-09-29T22:56:51.2555608Z Expected: 2
2023-09-29T22:56:51.2556137Z Actual:   1
2023-09-29T22:56:51.2556645Z   Stack Trace:
2023-09-29T22:56:51.2563218Z      at OpenTelemetry.Instrumentation.AspNetCore.Tests.BasicTests.DiagnosticSourceExceptionCallBackIsNotReceivedForExceptionsHandledInMiddleware() in /home/runner/work/opentelemetry-dotnet/opentelemetry-dotnet/test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:line 1001
2023-09-29T22:56:51.2564704Z --- End of stack trace from previous location ---

Reply:
These seem to be flaky tests or transient error, which are not specific to TFMs.

Reply:
I merged this PR but please continue to look into this. I'm worried that we might have stumbled onto a .NET 8 bug and if so we should report it back to the team ASAP!